framework/src/controller/channels/child_process_channel.js

```diff
@@ -50,18 +50,20 @@ class ChildProcessChannel extends BaseChannel {
 
 		this.rpcSocket.bind(this.rpcSocketPath);
 
-		return new Promise((resolve, reject) => {
-			this.busRpcClient.call(
-				'registerChannel',
-				this.moduleAlias,
-				this.eventsList.map(event => event.name),
-				this.actionsList.map(action => action.name),
-				{ type: 'ipcSocket', rpcSocketPath: this.rpcSocketPath },
-				(err, result) => {
-					if (err) return reject(err);
-					return resolve(result);
-				}
-			);
+		let timeout;
+
+		return Promise.race([
+			this._waitForAllToFinnish(),
+			this._rejectIfAtLeastOneFails(),
+			new Promise((resolve, reject) => {
+				timeout = setTimeout(async () => {
+					await this.cleanup();
+					// TODO: Review if logger.error might be useful.
+					reject(new Error('ChildProcessChannel setup timeout'));
+				}, 2000); // TODO: Get from constants
+			}),
+		]).finally(() => {
+			clearTimeout(timeout);
```

Comment:
If we remove the `this.cleanup()` from the timeout, the `finally` can be removed as well.